internal/connect/command_offset_update.go

```diff
@@ -0,0 +1,140 @@
+package connect
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	connectv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect/v1"
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+func (c *offsetCommand) newAlterCommand() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update a connector's offsets",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.update,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Annotations:       map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Update offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset update lcc-123456 --config-file config.json",
+			},
+			examples.Example{
+				Code: "confluent connect offset update lcc-123456 --config-file config.json --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().String("config-file", "", "JSON file containing connector offsets to set to.")
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagFilename("config-file", "json"))
+	cobra.CheckErr(cmd.MarkFlagRequired("config-file"))
+
+	return cmd
+}
+
+func (c *offsetCommand) update(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+
+	request, err := c.getAlterOffsetRequestBody(cmd)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
+	if err != nil {
+		return err
+	}
+
+	var offsetStatus connectv1.ConnectV1AlterOffsetStatus
+	table := output.NewTable(cmd)
+
+	retry.Retry(5*time.Second, 30*time.Second, func() error {
```

Comment:
Let's choose something more frequent than 5s. I would not want to use a CLI tool that takes 5 seconds to refresh

Reply:
Moved it to 1 sec